### PR TITLE
Add whitelist and blacklist to GitBranchName

### PR DESCRIPTION
**Pull request by Landerstraeten**
| Q             | A
| ------------- | ---
| Branch        | grumpy-seventies
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Documented?   | yes
| Fixed tickets | /

### CODE REVIEW

File: doc/tasks/git_branch_name.md [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
+number.
 
 ```yaml
 # grumphp.yml
 parameters:
     tasks:
         git_branch_name:
-            matchers:
-                Branch name must contain JIRA issue number: /JIRA-\d+/
+            whitelist:
```
Comment (janvernieuwe) [inline doc/tasks/git_branch_name.md]: Best explain that **ALL** whitelist patterns have to match.

File: doc/tasks/git_branch_name.md [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
+                - "/JIRA-\d+/"
+            blacklist:
```
Comment (janvernieuwe) [inline doc/tasks/git_branch_name.md]: And **NONE** of the blacklist patterns.

File: spec/Task/Git/BranchNameSpec.php [hunk 3]
```diff
@@ -107,4 +110,16 @@ function it_runs_with_detached_head_setting(RunContext $context, GrumPHP $grumPH
         $result = $this->run($context);
         $result->isPassed()->shouldBe(false);
     }
+
+    function it_runs_with_blacklisted_items(RunContext $context, GrumPHP $grumPHP, Repository $repository)
+    {
+        $grumPHP->getTaskConfiguration('git_branch_name')->willReturn([
+            'blacklist' => ['master'],
```
Comment (janvernieuwe) [inline spec/Task/Git/BranchNameSpec.php]: Maybe test with multiple blacklisted items, to see if one match results in a failure?

File: src/Task/Git/BranchName.php [hunk 4]
```diff
@@ -89,9 +79,31 @@ public function run(ContextInterface $context): TaskResultInterface
             return TaskResult::createFailed($this, $context, $message);
         }
 
-        foreach ($config['matchers'] as $rule) {
+        foreach ($config['blacklist'] as $rule) {
```
Comment (janvernieuwe) [inline src/Task/Git/BranchName.php]: Either one big try-catch, or just put the error in the array?

**Comment by Landerstraeten** [comment]
Thanks @janvernieuwe

**Pull request merged**